src/effects/backends/effectsbackendmanager.cpp

```diff
@@ -0,0 +1,115 @@
+    m_pNumEffectsAvailable = std::make_unique<ControlObject>(
+            ConfigKey("[Master]", "num_effectsavailable"));
+    m_pNumEffectsAvailable->setReadOnly();
+
+    addBackend(EffectsBackendPointer(new BuiltInBackend()));
+#ifdef __LILV__
+    addBackend(EffectsBackendPointer(new LV2Backend()));
+#endif
+}
+
+void EffectsBackendManager::addBackend(EffectsBackendPointer pBackend) {
+    VERIFY_OR_DEBUG_ASSERT(pBackend) {
+        return;
+    }
+
+    m_effectsBackends.insert(pBackend->getType(), pBackend);
+
+    QList<QString> backendEffects = pBackend->getEffectIds();
+    for (const QString& effectId : backendEffects) {
+        m_manifests.append(pBackend->getManifest(effectId));
+    }
+
+    m_pNumEffectsAvailable->forceSet(m_manifests.size());
+
+    std::sort(m_manifests.begin(),
+            m_manifests.end(),
+            EffectManifest::alphabetize);
+}
+
+const QList<EffectManifestPointer> EffectsBackendManager::getManifestsForBackend(
+        EffectBackendType backendType) const {
+    auto pBackend = m_effectsBackends.value(backendType);
+    VERIFY_OR_DEBUG_ASSERT(pBackend) {
+        return QList<EffectManifestPointer>();
+    }
+    auto list = pBackend->getManifests();
+    std::sort(list.begin(), list.end(), EffectManifest::alphabetize);
+    return list;
+}
+
+EffectManifestPointer EffectsBackendManager::getManifestFromUniqueId(
+        const QString& uid) const {
+    if (kEffectDebugOutput) {
+        //qDebug() << "EffectsBackendManager::getManifestFromUniqueId" << uid;
+    }
+    if (uid.isEmpty()) {
+        // Do not DEBUG_ASSERT, this may be a valid request for a nullptr to
+        // unload an effect.
+        return EffectManifestPointer();
+    }
+    int delimiterIndex = uid.lastIndexOf(" ");
+    EffectBackendType backendType =
+            EffectManifest::backendTypeFromString(uid.mid(delimiterIndex + 1));
+    VERIFY_OR_DEBUG_ASSERT(backendType != EffectBackendType::Unknown) {
+        // Mixxx 2.0 - 2.2 did not store the backend type in mixxx.cfg,
+        // so this code will be executed once when upgrading to Mixxx 2.3.
+        // This debug assertion is safe to ignore in that case. If it is
+        // triggered at any later time, there is a bug somewhere.
+        // Do not manipulate the string passed to this function, just pass
+        // it directly to BuiltInBackend.
+        return m_effectsBackends.value(EffectBackendType::BuiltIn)
+                ->getManifest(uid);
+    }
+    return m_effectsBackends.value(backendType)
+            ->getManifest(uid.mid(-1, delimiterIndex + 1));
+}
+
+EffectManifestPointer EffectsBackendManager::getManifest(
+        const QString& id, EffectBackendType backendType) const {
+    return m_effectsBackends.value(backendType)->getManifest(id);
+}
+
+const QString EffectsBackendManager::getDisplayNameForEffectPreset(
+        EffectPresetPointer pPreset) const {
+    QString displayName(QObject::tr("None"));
```

Comment:
can you factor out "None" to a translated expression somewhere?  It'll be hard for translators to understand what is meant here.

Reply:
I can add a translator tip.